pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -229,7 +229,9 @@ public ConfigSuccessResponse addTable(String tableConfigStr,
       TableConfigTunerUtils.applyTunerConfigs(_pinotHelixResourceManager, tableConfig, schema, Collections.emptyMap());
 
       // TableConfigUtils.validate(...) is used across table create/update.
-      TableConfigUtils.validate(tableConfig, schema, typesToSkip, _controllerConf.isDisableIngestionGroovy());
+      TableConfigUtils.validate(tableConfig, schema, typesToSkip, _controllerConf.isDisableIngestionGroovy(),
```

Comment:
`disableGroovy` and this new check is always instance level. We don't want to keep adding new booleans into the validate. We can make them static field, and initialize them when the controller starts.

Reply:
I'd suggest keeping it consistent with disable ingestion groovy, and we can make it instance level for now. In the future we can revisit this if necessary